language-plutus-core/src/Language/PlutusCore/Parser.y

```diff
@@ -139,44 +139,28 @@ app loc t (t' :| ts) = Apply loc (app loc t (t':|init ts)) (last ts)
 
 handleInteger :: AlexPosn -> Natural -> Integer -> Parse (Constant AlexPosn)
 handleInteger x sz i = if isOverflow
-    then throwE (Overflow x sz i)
+    then throwError (Overflow x sz i)
     else pure (BuiltinInt x sz i)
 
     where isOverflow = i < (-k) || i > (k - 1)
           k = 8 ^ sz `div` 2
 
-parseST :: BSL.ByteString -> Either ParseError (IdentifierState, Program TyName Name AlexPosn)
-parseST str = liftErr (runAlexST str (runExceptT parsePlutusCore))
-    where liftErr (Left s)  = Left (LexErr s)
-          liftErr (Right x) = normalize x
-          normalize (x, Left y) = Left y
-          normalize (x, Right y) = Right (x, y)
+parseST :: BSL.ByteString -> StateT IdentifierState (Except ParseError) (Program TyName Name AlexPosn)
```

Comment:
`QT (Except ParseError) (Program TyName Name AlexPosn)`?

Reply:
This is not a problem of this PR, but I do not like that `IdentifierState` is thrown away in case `ParseError` is returned. It can be useful for debugging.

Reply:
I could re-jigger this so that `IdentifierState` is kept as well if you'd like.

Reply:
That would be better I think.

Reply:
I wasn't quite brave enough to pipe this the whole way through - in particular this then goes on to affect `parseScoped` and the things that use that.

Reply:
I also don't know how to put `parseScoped` into this properly, since it uses `rename`, which doesn't produce a modified `IdentifierState` :(

Reply:
That's not too important, I think.